### PR TITLE
Add get and set functions for the nametag color

### PR DESCRIPTION
**Pull request by TeTpaAka**
Let's you set the color of the nametag of a player by
player:set_nametag_color({ a = 0, r = 0, g = 0, b = 0 })

And you can get the current color with
local color = player:get_nametag_color()

Fixes https://github.com/minetest/minetest/issues/1345, at least for the part of color and hiding.

### CODE REVIEW

**Comment by est31** [comment]
very nice, nametag API is very a very wanted feature by users, I'll review later today.

File: src/content_cao.cpp [hunk 1]
```diff
@@ -1715,6 +1716,12 @@ void GenericCAO::processMessage(const std::string &data)
 			m_armor_groups[name] = rating;
 		}
 	}
+	else if (cmd == GENERIC_CMD_SET_NAMETAG_COLOR) {
```
Comment (est31) [inline src/content_cao.cpp]: `}` on same line please.

File: src/script/lua_api/l_object.cpp [hunk 2]
```diff
@@ -1274,6 +1274,48 @@ int ObjectRef::l_override_day_night_ratio(lua_State *L)
...
 
+// set_nametag_color(self, color)
+int ObjectRef::l_set_nametag_color(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	PlayerSAO *playersao = getplayersao(ref);
+	if (playersao == NULL)
+		return 0;
+
```
Comment (est31) [inline src/script/lua_api/l_object.cpp]: please remove the trailing whitespace.

File: src/script/lua_api/l_object.cpp [hunk 3]
```diff
@@ -1274,6 +1274,48 @@ int ObjectRef::l_override_day_night_ratio(lua_State *L)
...
+// get_nametag_color(self)
+int ObjectRef::l_get_nametag_color(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkobject(L, 1);
+	PlayerSAO *playersao = getplayersao(ref);
+	if (playersao == NULL)
+		return 0;
+
```
Comment (est31) [inline src/script/lua_api/l_object.cpp]: please remove the trailing whitespace.

**Comment by est31** [comment]
See comments above, otherwise :+1:

**Comment by TeTpaAka** [comment]
Fixed.

**Comment by Novatux** [comment]
:+1:

**Comment by est31** [comment]
86a963caca9604ad57904e9acd9bef7c46ca47d8

**Comment by neoascetic** [comment]
Great change, thanks!
However, I've faced with issue: I want to make nametags invisible by setting alpha channel to 0. And this almost works, but grey font's shadow of a nametag is still visible. How could I remove it (or is there any other way to make nametag invisible)? Thanks.

**Comment by est31** [comment]
In theory I have coded a change to fix this: d2ca662569427d36642660314668e416bf68f3c8
Does it still appear with up to date minetest?

**Comment by neoascetic** [comment]
Yeah, a shadow is still visible.

**Pull request closed without merging**